test/unit/services/s3.test.js

```diff
@@ -80,3 +86,138 @@ describe('listObjects', () => {
+    LastModified: '2022-07-27T03:17:56.000Z',
+    ETag: '"4727374fff1eb029687319184bc23a10"',
+    ChecksumAlgorithm: [],
+    Size: 1761676,
+    StorageClass: 'STANDARD',
+    Body: 'sample buffer',
+  }
+  const testGetObject = jest.fn().mockReturnValue({
+    promise: () => Promise.resolve(sampleS3Object),
+  });
+  const s3 = {
+    getObject: testGetObject,
+  };
+  const { Key: sampleKey } = sampleS3Object;
+
+  // test
+  const getObjectResult = await getObject(sampleKey, s3);
+
+  // asssert
+  expect(getObjectResult).toEqual(sampleS3Object.Body)
+  });
+})
+
+describe('moveObject', () => {
+  test('Calls s3.copyObject', async () => {
+
+    // set up mocks
+    const deleteObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+    const copyObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+
+    const s3 = {
+      copyObject,
+      deleteObject,
+    }
+
+    //test
+    const result = await moveObject('test-origin-folder/test-s3-object.png', 'test-origin-folder/', 'test-destination-folder/', s3)
+    
+    // assert
+    expect(copyObject).toHaveBeenCalled();
+  });
+
+  test('Calls s3.deleteObject', async () => {
+
+    // set up mocks
+    const deleteObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+    const copyObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+
+    const s3 = {
+      copyObject,
+      deleteObject,
+    }
+
+    //test
+    const result = await moveObject('test-origin-folder/test-s3-object.png', 'test-origin-folder/', 'test-destination-folder/', s3)
+
+    // assert
+    expect(deleteObject).toHaveBeenCalled();
+  });
+})
+
+describe('moveObjects', () => {
+  test('Calls moveObject on every item in an array', async () => {
+    const testArray = [
+      {
+        Key: 'staging/IMG_20161216_142249.jpg',
+        LastModified: '2022-07-27T03:17:56.000Z',
+        ETag: '"4727374fff1eb029687319184bc23a10"',
+        ChecksumAlgorithm: [],
+        Size: 1761676,
+        StorageClass: 'STANDARD',
+        Body: 'sample buffer',
+      },
+      {
+        Key: 'staging/IMG_20161216_142249.jpg',
+        LastModified: '2022-07-27T03:17:56.000Z',
+        ETag: '"4727374fff1eb029687319184bc23a10"',
+        ChecksumAlgorithm: [],
+        Size: 1761676,
+        StorageClass: 'STANDARD',
+        Body: 'sample buffer',
+      }
+    ]
+
+    // set up mocks
+    const deleteObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+    const copyObject = jest.fn().mockReturnValue({
+      promise: () => Promise.resolve(),
+    });
+
+    const s3 = {
+      copyObject,
+      deleteObject,
+    }
+
+    //test
+    const result = await moveObjects(testArray, 'test-origin-folder/', 'test-destination-folder/', s3)
+
+    // assert
+    expect(copyObject).toHaveBeenCalledTimes(2);
+    expect(deleteObject).toHaveBeenCalledTimes(2);
```

Comment:
Looks great! I say these assertions are good as-is because this function is concerned about calling the expected `s3` functions the right amount of times